### ISSUE TITLE
What is the difference between mediacodec and mediacodec-copy aka hw vs hw+?

### ISSUE THREAD

**Issue by rohitghali** [mpv-android/mpv-android#623]
I've looked everywhere and couldn't find a proper answer. I used hw+ decoding in MX player all these years without even knowing what the difference is. Now, that mpv added it too, I want to know the clear difference, if possible, in layman terms please. How do they fare in terms of performance and video quality?

**Comment by lavilao** [comment]
I guess the same difference between vaapi and vaapi-copy on linux, which is basically the frame doesnt get copied back to system memory from gpu. This improves performance but it makes mpv unable to alter the vídeo with filters (unless they are hw filters)

**Comment by sfan5** [comment]
Correct.
Worth noting that I have no idea what "HW+" means in MXPlayer and it may have nothing to do with what mpv does.
